### PR TITLE
Fix incognito mode.

### PR DESCRIPTION
**Pull request by RadiantUwU**
This pull request makes the incognito mode back visible, allows mods to get access to the list of incognito players, and makes incognito mode work yet again.

### CODE REVIEW

File: MainModule/Server/Commands/HeadAdmins.lua [hunk 1]
```diff
@@ -431,73 +431,61 @@ return function(Vargs, env)
...
 							if plr then
 								if not table.find(service.IncognitoPlayers, plr) then
 									table.insert(service.IncognitoPlayers, plr)
 								end
-
-								plr:Remove()
+								if plr.Character then
+									plr.Character:Destroy()
+								end
+								plr:Destroy()
```
Comment (ccuser44) [inline MainModule/Server/Commands/HeadAdmins.lua:488]: This is just wrong

You should not delete the character, use `:invis`

**Review by ccuser44 — CHANGES_REQUESTED** [review]

File: MainModule/Server/Commands/HeadAdmins.lua (hunk 1 above)
Comment (RadiantUwU) [reply MainModule/Server/Commands/HeadAdmins.lua:488]: This does not hide the player from the leader board.

Comment (ccuser44) [reply MainModule/Server/Commands/HeadAdmins.lua:488]: Oh sorry I misinterpreted it partly wrong, I agree it should persist but I disagree with the character deletion

The character shouldn't be deleted

**Comment by Dimenpsyonal** [comment]
What's the rationale behind moving the list from admins to mods

**Comment by RadiantUwU** [comment]
> What's the rationale behind moving the list from admins to mods

There is now an argument whether to hide only from nonadmins or from everyone else
So with this, if you hide from everyone you will be also hidden from the list, if not you will not be hidden from mods, so why not give the mods the list of the players that they can anyways still see?

File: MainModule/Server/Commands/HeadAdmins.lua (hunk 1 above)
Comment (RadiantUwU) [reply MainModule/Server/Commands/HeadAdmins.lua:488]: > Oh sorry I misinterpreted it partly wrong, I agree it should persist but I disagree with the character deletion
> 
> The character shouldn't be deleted

Why is that? Deleting the character is a very easy thing and won't cause complications doing so.

**Comment by Dimenpsyonal** [comment]
I suppose so yea

File: MainModule/Server/Commands/HeadAdmins.lua (hunk 1 above)
Comment (ccuser44) [reply MainModule/Server/Commands/HeadAdmins.lua:488]: Oh sorry I misinterpreted it partly wrong, I agree it should persist but I disagree with the character deletion

The character shouldn't be deleted

> > Oh sorry I misinterpreted it partly wrong, I agree it should persist but I disagree with the character deletion
> > The character shouldn't be deleted
> 
> Why is that? Deleting the character is a very easy thing and won't cause complications doing so.

What if you want the character to exist though? The command is supposed to remove the player not the character

Just a theoretical example: If I want to troll people to fake myself to be an NPC in the game when I'm not

**Comment by RadiantUwU** [comment]
@ccuser44 Please re-review this.

**Comment by ccuser44** [comment]
I still am not suree why the character deletion should be a part of the incognito command

You can use invis command instead

**Comment by RadiantUwU** [comment]
> I still am not suree why the character deletion should be a part of the incognito command
> 
> You can use invis command instead

this would honestly also help against esp hacks

**Comment by RadiantUwU** [comment]
Tested, requires a few bug fixes

**Comment by RadiantUwU** [comment]
Works as expected, however was unable to test the case of an admin user joining while someone is invis, i was able to test it on a non admin, however i think that test wont be necessary Loader/MainModule/Server/Core/Process.lua:1022 ```lua
					if level > 0 and not pdata.hide_from_everyone then
						continue
					end
```

**Comment by ccuser44** [comment]
> > I still am not suree why the character deletion should be a part of the incognito command
> > 
> > You can use invis command instead
> 
> this would honestly also help against esp hacks

It still shouldnt be in the incognito command

**Comment by RadiantUwU** [comment]
> It still shouldnt be in the incognito command

I'm still waiting for a reason why.

**Comment by RadiantUwU** [comment]
> There is no good reason to have it in

Yes there is, it hides the player by default, not having to run another command to do it.

**Comment by RadiantUwU** [comment]
Proof of functionality has been satisfied by me testing it in studio.

**Comment by Dimenpsyonal** [comment]
Cuz im waiting to see whether invis will be included

**Comment by RadiantUwU** [comment]
> Cuz im waiting to see whether invis will be included

invis is already included *inside* the command as can be seen from here
```lua
Args = {"player", "hideFromNonAdmins(default true)", "hideCharacter(default true)"};
```

**Comment by RadiantUwU** [comment]
> Cuz im waiting to see whether invis will be included

What do you want me to do exactly? use :invis inside the command and do that instead of deleting the character?

**Comment by Dimenpsyonal** [comment]
Just waiting to see what ccuser says

**Comment by Dimenpsyonal** [comment]
> Proof of functionality has been satisfied by me testing it in studio with a local server.

yea but is there like a video or a screenshot or something

**Comment by RadiantUwU** [comment]
> > Proof of functionality has been satisfied by me testing it in studio with a local server.
> 
> yea but is there like a video or a screenshot or something

It's gonna be quite a pain to get them as i use Linux and roblox does *not* like linux.

**Comment by Dimenpsyonal** [comment]
Unfortunate

**Review by BLADE649 — CHANGES_REQUESTED** [review]
Idk really know how to put on roblox how?

**Pull request merged**